include/tvm/runtime/threading_backend.h

```diff
@@ -0,0 +1,31 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file threading_backend.h
+ * \brief Utilities for manipulating thread pool threads.
+ */
+#ifndef TVM_RUNTIME_THREADING_BACKEND_H_
+#define TVM_RUNTIME_THREADING_BACKEND_H_
+
+namespace tvm {
+namespace runtime {
+namespace threading {
+
+class ThreadGroup {
+  public:
+    virtual void Launch(std::vector<std::function<void()>> task_callbacks) =0;
```

Comment:
Could this instead be``` std::function<void(int task_id)>```, works for most case and can adapt to the task_callback interface if wanted.

Reply:
Since we can only launch once, this can be part of Create Function

Reply:
Ah, that's the thing: the current interface doesn't imply only one launch. One could do `Launch(tasks)` and then `Launch(some_more_tasks)`. That's why it's expected for the caller to explicitly bind their own `task_id` to the callback before passing it in.

Reply:
So there are two kind of possibility here, if we want the maximum efficiency through a persistent thread group and multiple launches, then the std::vector<std::function> is a bit too much in here.

If we don't care that much about efficiency, then have only one launch per thread group works fine and we can create multiple thread groups for different launch